src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp

```diff
@@ -1430,6 +1430,41 @@ Error executeRcppEngineChunk(const std::string& docId,
    return error;
 }
 
+Error stanExecutionError(const std::string& docId,
+                         const std::string& chunkId,
+                         const std::string& message,
+                         const FilePath& targetPath,
+                         const ErrorLocation& errorLocation)
+{
+   // emit chunk error
+   chunkConsoleOutputHandler(
+            module_context::ConsoleOutputError,
+            message,
+            targetPath);
+   
+   // forward failure to chunk
+   notebook::enqueueChunkOutput(
+            docId,
+            chunkId,
+            notebook::notebookCtxId(),
+            kChunkOutputText,
+            targetPath);
+   
+   return Error(r::errc::CodeExecutionError, errorLocation);
+}
+
+Error stanExecutionError(const std::string& docId,
+                         const std::string& chunkId,
+                         const FilePath& targetPath,
+                         const ErrorLocation& errorLocation)
+{
+   std::string message =
+         "engine.opts$x must be a character string providing a "
+         "name for the returned `stanmodel` object";
+         
+   return stanExecutionError(docId, chunkId, message, targetPath, errorLocation);
```

Comment:
For stanExecutionError, it looks to me as if the error will be reported inline within the notebook (great) but the Error itself will also get propagated to the RPC caller, which then might (depending on how the client side is coded) show an error dialog.I think that if we execute `chunkConsoleOutputHandler` and `notebook::enqueueChunkOutput` then the overall function invocation is actually a Success (the error is between the user and their code).

At the same time we certainly might want to log some errors that we consider "impossible" (e.g. `base::assign` working). So overall I'd say that this function should be more like `reportStanError` and it shouldn't return the error. Then callers would do this:

```cpp
if (error) {
    // possibly log depending on what the error was
    reportStanError(error);
    return Success();
}
```